src/fknni/faiss/faiss.py

```diff
@@ -32,87 +31,156 @@ def __init__(
                       where closer neighbors have a higher influence on the imputation.
             index_factory: Description of the Faiss index type to build. Defaults to 'Flat'.
         """
-        super().__init__()
+        if n_neighbors < 1:
+            raise ValueError("n_neighbors must be at least 1.")
+        if strategy not in {"mean", "median", "weighted"}:
+            raise ValueError("Unknown strategy. Choose one of 'mean', 'median', 'weighted'")
+
         self.missing_values = missing_values
         self.n_neighbors = n_neighbors
         self.metric = metric
         self.strategy = strategy
         self.index_factory = index_factory
+        self.min_data_ratio = 0
+        self.X_full = None
+        self.features_nan = None
+        self.min_data_ratio = 0.25
+        super().__init__()
 
-    def fit(self, X: np.ndarray | pd.DataFrame, *, y: np.ndarray | None = None) -> FaissImputer:
-        """Fits the FaissImputer to the provided data.
+    @overrides
+    def fit_transform(self, X: np.ndarray, y=None, **fit_params) -> ndarray[Any, dtype[Any]] | None:
+        """Imputes missing values in the data using the fitted Faiss index.
 
         Args:
             X: Input data with potential missing values.
             y: Ignored, present for compatibility with sklearn's TransformerMixin.
 
-        Raises:
-          ValueError: If any parameters are set to an invalid value.
+        Returns:
+            Data with imputed values as a NumPy array of the original data type.
         """
-        X = np.asarray(X, dtype=np.float32)
-        if isinstance(X, pd.DataFrame):
-            X = X.replace(self.missing_values, np.nan).values
-        else:
-            X = np.where(X == self.missing_values, np.nan, X)
-
-        if np.isnan(X).all(axis=0).any():
+        self.X_full = np.asarray(X, dtype=np.float32)
```

Comment:
I generally wonder whether we should change the dtype of the input matrix. If a user gives float64, should we keep it as float64 and also return that?